whole-tale/ngx-dashboard
# Render Tale Metadata when displaying "Version Info" instead of raw JSON

## Problem
Viewing a Tale "Version Info" only shows raw JSON.

Fixes #144 

## Approach
Show rendered Tale metadata from the selected version instead of raw JSON.

* Add new API call to `TaleService` for `GET /tale/{id}/restore`
* Refactor `tale-metadata-tab` to isolate the readonly part into a new component called `rendered-tale-metadata`
* Replace content of TaleVersionInfoDialogComponent to display the new `rendered-tale-metadata` component

## How to Test
Prerequisites: at least on Tale created that you own

1. Checkout this branch locally (along with https://github.com/whole-tale/wt_versioning/pull/24), rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Navigate to the Run View for a Tale that you own
4. Create a version of the Tale (if you haven't already)
5. Expand the dropdown beside the new version and choose "View Version Info"
    * You should see the TaleVersionInfoDialogComponent appear
    * You should see a rendered copy of the Tale version's metadata is shown
    * You should see that all fields appear in the same order as in the Run Tale > Metadata view
6. Add an author to the Tale and save a new version
7. "View Version Info" for the new version you created, compare with old "Version Info" contents
    * You should see that the author you added appears here, but does not appear in the previous "Version Info"
